src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+
+import scala.annotation.tailrec
+
+/**
+  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
+  * approximately balanced loads among processors.
+  */
+object SeedJointSelector {
+
+  /**
+    * Searches though sequence of joint sets to find best seed joints
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numProcessors Number of processors to be used in the analysis
+    * @return Returns a Seq containing the seed joints that best divide the input volume. If unable to find
+    *         satisfactory seed joints, returns None.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block,
+                      numProcessors: Int): Option[Seq[Joint]] = {
+    if (jointSets.isEmpty) {
+      println("ERROR: Unable to find seed joints")
+      None
+    } else if (jointSets.head(0).shape.isEmpty) {
+      // Check that joint set is persistent
+      val seedJointOption = findSeedJoints(jointSets.head, inputVolume, numProcessors, inputVolume.volume)
+
+      if (seedJointOption.isEmpty) {
+          searchJointSets(jointSets.tail, inputVolume, numProcessors)
+      } else {
+        seedJointOption
+      }
+    } else {
+      searchJointSets(jointSets.tail, inputVolume, numProcessors)
+    }
+  }
+
+  /**
+    * Cycles through persistent joint set to find joints that approximately balances load across processes
+    *
+    * @param jointSet Seq of Joints representing a persistent joint set
+    * @param rockVolume Block representing input rock volume that should be divided into approximately equal pieces
+    *                   for each process to operate on
+    * @param numProcessors Number of processes that will be used in analysis
+    * @param totalVolume Total volume of rockVolume
+    * @param tolerance Tolerance in deviation from ideal volume per processor. Default initial value is 1%, though it
+    *                  will be updated as necessary in recursive calls.
+    * @param stepSize Step size that tolerance will be incremented by if unable to find enough seed joints. Default
+    *                 initial value is 1%, though it will be updated as necessary in recursive calls.
+    * @return If able to find the required number of seed joints, this function will return a Seq of Joints. If unable
+    *         to find sufficient seed joints, it will return None.
+    */
+  @tailrec
+  private def findSeedJoints(jointSet: Seq[Joint], rockVolume: Block, numProcessors: Integer, totalVolume: Double,
+                     tolerance: Double = 0.01, stepSize: Double = 0.01): Option[Seq[Joint]] = {
+    if (jointSet.length < numProcessors - 1) {
+      println(s"Error: Not enough joints to generate required number of seed joints. ${jointSet.length} joints but"+
+      s" $numProcessors processors. Try using less processors.")
```

Comment:
Well since I've already been picky about code in this review, why not add English to the mix? I think it would be better to say "Try using fewer processors". This is really not that important though :)